evm/src/main/java/org/hyperledger/besu/evm/precompile/P256VerifyPrecompiledContract.java

```diff
@@ -222,7 +222,14 @@ private PrecompileInputResultTuple computeDefault(final Bytes input) {
       return new PrecompileInputResultTuple(
           enableResultCaching ? input.copy() : input, PrecompileContractResult.success(INVALID));
     }
-
+    try {
+      final org.bouncycastle.math.ec.ECPoint ecPoint = R1_PARAMS.getCurve().createPoint(qx, qy);
+      signatureAlgorithm.getCurve().validatePublicPoint(ecPoint);
```

Comment:
This uses boringSSL validation presumably? We could use bouncy castle's as well technically from here: https://github.com/bcgit/bc-java/blob/b41f23936724284a20f10dff13c76896a846031b/core/src/main/java/org/bouncycastle/math/ec/ECCurve.java#L139-L147 

since the default pathway is enabled when boringssl is not available?

Reply:
This code path is only used in computeDefault() which is what is used when boringSSL is disabled.  

This code path is actually bouncycastle.  R1_PARAMS.getCurve() returns ECCurve configured for 
'secp256r1'.